benches/indexed_crate.rs

```diff
@@ -0,0 +1,14 @@
+use divan::Bencher;
+use trustfall_rustdoc_adapter::IndexedCrate;
+
+fn main() {
+    divan::main()
+}
+
+#[divan::bench]
+fn new(b: Bencher) {
+    let data = std::fs::read_to_string("/home/jalil/Dev/aws-sdk-rust/target/doc/aws_sdk_ec2.json")
```

Comment:
I have a few concerns and a few suggestions on how to address them.

At a high level, the next person looking at this code in 6 months will probably not understand most of the context here. They won't know why this file was chosen, why that was a sound decision (as opposed to choosing any other file), how to create it, and almost certainly won't have a `/home/jalil/Dev` directory to place it in.

It would be good to automate the bits that can be automated, and thoroughly document the rest.

A better setup here would be:
- The benchmarking function runs a series of commands that, starting with "a machine with an internet connection," result in this file being available in a temp directory. If the file was already available (e.g. from a previous run), then it is reused as-is.
- The module uses a module-level doc comment (`//!` prefix starting on line 1) to describe why this crate's rustdoc JSON in particular is chosen: because it is very large, and highly representative of the perf environment facing the largest users of `trustfall-rustdoc-adapter`.

An example of the automated flow of commands could be:
- Check if the system's temp directory (whatever the OS!) contains a subdirectory called `aws-sdk-rust`. If it doesn't, `git clone` the relevant repo into such a subdirectory.
- Check out a specified commit hash (e.g. the current `HEAD` of the main branch) in that directory.
- Run a specified cargo command in that directory, causing the expected rustdoc JSON file to be created.
- If the specified commit hash was already checked out, and the expected rustdoc JSON file appeared to already be present at the expected location, skip the `git checkout` and `cargo` invocations and assume the file is good to reuse. *Document this behavior* in the module-level comment.
- Then, begin the benchmark using that file.

Reply:
Do we want to keep the benchmarks? I added them in a separate commit thinking I would drop it before marking it ready to merge (that's why they are in that state).

If we want them, I'll spend more time polishing them!

Reply:
If you're open to polishing to the point where we merge them, that'd be sweet! Having nice, maintainable benchmarks will make it easier to merge future performance improvements both here and in `cargo-semver-checks`. We'll be able to make more ambitious changes on the basis of what we measure, as opposed to just going on intuition.

But if you don't want to do that work, I'd merge the PR without them too.

Reply:
I wrote my thoughts on the commit introducing benchmarks, but they are polished enough to merge rn